### PR TITLE
Re-wire indexing configuration

### PR DESCRIPTION
**Pull request by prateek**

### CODE REVIEW

File: services/m3dbnode/config/config_test.go [hunk 1]
```diff
@@ -186,6 +186,16 @@ pooling:
         capacity: 4096
         lowWatermark: 0.01
         highWatermark: 0.02
+    tagArrayPool:
+        size: 4096
```
Comment (robskillington) [inline services/m3dbnode/config/config_test.go]: Might want to use a little higher considering this is kind of like an example config.  Perhaps 65536?

File: services/m3dbnode/config/config_test.go [hunk 2]
```diff
@@ -186,6 +186,16 @@ pooling:
...
+        capacity: 8
+        maxCapacity: 32
+        lowWatermark: 0.01
+        highWatermark: 0.02
+    indexResultsPool:
+        size: 4096
```
Comment (robskillington) [inline services/m3dbnode/config/config_test.go]: Probably won't be more than 4k distinct queries at any one point in time I guess, 8192 might be safer though.

File: services/m3dbnode/config/config_test.go (hunk 1 above)
Comment (prateek) [reply services/m3dbnode/config/config_test.go]: Sure thing

File: services/m3dbnode/config/config_test.go (hunk 2 above)
Comment (prateek) [reply services/m3dbnode/config/config_test.go]: Sure thing

Comment (prateek) [reply services/m3dbnode/config/config_test.go]: done

File: services/m3dbnode/config/config_test.go (hunk 1 above)
Comment (prateek) [reply services/m3dbnode/config/config_test.go]: done

**Pull request merged**